TP8 Documentation and Release Notes Updates

geekspertise opened the pull request:
Here are the updated docs for Service Mesh TP8 including the 3scale Istio Adapter 0.3 documentation update and some general clean up. Can @knrc @tvieira @JStickler @brian-avery and @vramosp provide feedback?

knrc commented:
@geekspertise These changes look good to me, is there a preview version?

geekspertise commented:
> @geekspertise These changes look good to me, is there a preview version?

Unfortunately the OpenShift Preview Bot doesn't seem like it was ever fixed to work with my account.

tvieira commented:
LGTM

JStickler commented on servicemesh-install/topics/appendix-threescale-adapter.adoc:
```diff
@@ -138,9 +137,86 @@ Caching can be disabled entirely by setting `cacheEntriesMax` to a non-positive
+
+If you want the adapter to examine a different query parameter or request header, change the name as appropriate. For example, to check for the app ID in a query parameter named “identification”, change `request.query_params["app_id"]` to `request.query_params["identification"]`.
+
+=== Hybrid pattern
```
Hybrid authentication method

geekspertise commented:
PR Update w/ fresh content: 
@knrc and I added a new topic comparing the upstream Istio community install versus Service Mesh. I also addressed the previous comments and a few other issues. @knrc and @JStickler can you provide feedback on the new content? And @vramosp please take a look at the 3scale Adapter docs as some language changes were made so it fits with the Service Mesh docs.

brian-avery commented on servicemesh-install/topics/install-remove.adoc:
```diff
@@ -1,7 +1,7 @@
 = Removing {ProductName}
 
-The following command removes the {ProductShortName} from an existing installation.  You can run these commands from any host with access to the cluster.  Ensure you are logged in as a cluster admin before running the following commands.
+The following command removes the {ProductShortName} from an existing installation. You can run these commands from any host with access to the cluster. Ensure you are logged in as a cluster admin before running the following commands.
```
The following command removes...

...before running the following commands.

Seems repetitive.

The following command removes the {ProductShortName} from an existing installation. Provided you are logged in as a cluster admin, you can run these commands from any host with access to the cluster. ?

brian-avery commented on servicemesh-install/topics/servicemesh-overview.adoc:
```diff
@@ -8,9 +8,8 @@ This release of {ProductName} is a Technology Preview release only.  Technology
-Based on the  open source https://istio.io/[Istio] project, {ProductName}  adds a transparent layer on existing distributed applications without requiring any changes to the service code.  You add {ProductName}  support to services by deploying a special sidecar proxy throughout your environment that intercepts all network communication between microservices.   You configure and manage the servic ...
-
-{ProductName} provides an easy way to create a network of deployed services that provides discovery, load balancing, service-to-service authentication, failure recovery, metrics, and monitoring.   A service mesh also provides more complex operational functionality, including A/B testing, canary releases, rate limiting, access control, and end-to-end authentication.
+Based on the open source https://istio.io/[Istio] project, {ProductName}  adds a transparent layer on existing distributed applications without requiring any changes to the service code. You add {ProductName} support to services by deploying a special sidecar proxy throughout your environment that intercepts all network communication between microservices. You configure and manage the service mes ...
```
{ProductName} adds has two spaces.

brian-avery commented on servicemesh-install/topics/tutorial-prometheus.adoc:
```diff
@@ -1,7 +1,7 @@
 = Prometheus tutorial
 
-Prometheus is an open source systems and service monitoring toolkit.  Prometheus collects metrics from configured targets at specified intervals, evaluates rule expressions, displays the results, and can trigger alerts if some condition is observed to be true.  Grafana or other API consumers can be used to visualize the collected data.
+Prometheus is an open source systems and service monitoring toolkit.  Prometheus collects metrics from configured targets at specified intervals, evaluates rule expressions, displays the results, and can trigger alerts if some condition is observed to be true. Grafana or other API consumers can be used to visualize the collected data.
```
Prometheus is an open source system

brian-avery commented on servicemesh-release-notes/topics/servicemesh-overview.adoc:
```diff
@@ -8,9 +8,8 @@ This release of {ProductName} is a Technology Preview release only.  Technology
-Based on the  open source https://istio.io/[Istio] project, {ProductName}  adds a transparent layer on existing distributed applications without requiring any changes to the service code.  You add {ProductName}  support to services by deploying a special sidecar proxy throughout your environment that intercepts all network communication between microservices.   You configure and manage the servic ...
-
-{ProductName} provides an easy way to create a network of deployed services that provides discovery, load balancing, service-to-service authentication, failure recovery, metrics, and monitoring.   A service mesh also provides more complex operational functionality, including A/B testing, canary releases, rate limiting, access control, and end-to-end authentication.
+Based on the open source https://istio.io/[Istio] project, {ProductName}  adds a transparent layer on existing distributed applications without requiring any changes to the service code. You add {ProductName} support to services by deploying a special sidecar proxy throughout your environment that intercepts all network communication between microservices. You configure and manage the service mes ...
```
There are two spaces after {ProductName}

knrc commented on servicemesh-install/topics/servicemesh-istio-comparison.adoc:
```diff
@@ -0,0 +1,45 @@
+The upstream Istio community installation automatically injects the sidecar to namespaces you have labeled.
+
+{ProductName} does not automatically inject the sidecar to any namespaces, but requires you to specify sidecar injection in the https://docs.openshift.com/container-platform/3.11/servicemesh-install/servicemesh-install.html#updating-master-configuration[master configuration].
+
```
The automatic injection should refer to the annotation rather than configuration of the webhooks, IIRC we have a section in the documentation which has the text so we could refer to that section if necessary.

knrc commented on servicemesh-install/topics/servicemesh-istio-comparison.adoc:
```diff
@@ -0,0 +1,45 @@
+== Role Based Access Control features
+Role Based Access Control (RBAC) provides a mechanism you can use to control access to a service. You can identify subjects by username or by specifying a set of properties and apply access controls accordingly.
+
+The upstream Istio community installation includes options to perform exact matches, or check for a header containing a specific prefix or suffix.
```
Sorry, there should also be a wildcard match in this list

exact, wildcard, prefix and suffix.

knrc commented on servicemesh-install/topics/servicemesh-istio-comparison.adoc:
```diff
@@ -0,0 +1,45 @@
+      request.headers[<header>]: "value"
+----
+
+{ProductName} extends the ability to match request headers by using a regular expression. Specify a property key of `request.regex.headers` with a valid, regular expression.
```
Should the comma be there after valid?

knrc commented on servicemesh-install/topics/servicemesh-istio-comparison.adoc:
```diff
@@ -0,0 +1,45 @@
+
+{ProductName} extends the ability to match request headers by using a regular expression. Specify a property key of `request.regex.headers` with a valid, regular expression.
+
+.{ProductName} matching request headers example
```
Perhaps include 'regular expression' in this?

JStickler commented on servicemesh-release-notes/topics/rn-known-issues.adoc:
```diff
@@ -71,16 +71,14 @@ pass:quotes[*kubeConfigFile: /dev/null*]
 Create a soft link from `/var/lib/origin/null` to `/dev/null` and replace the two kubeConfigFile variables to `/var/lib/origin/null`. This makes the task think /var/lib/origin/null is an asset in the origin "perimiter" and continue running.
 
-https://issues.jboss.org/browse/MAISTRA-62[MAISTRA-62] After a successful Istio installation, when upgrading an OCP cluster from 3.10 to 3.11, the storage upgrade task fails causing the entire upgrade process to fail.
+https://issues.jboss.org/browse/MAISTRA-245[MAISTRA-245] The sidecar injector pod fails to start if you are running the upstream community version.
```
It looks like there is a workaround for this issue that should also be documented.

geekspertise commented:
@kalexand-rh and @vikram-redhat I got the blessing from @knrc to merge this content.

vikram-redhat commented:
@geekspertise - I can't recall if we are going through the peer review process for ServiceMesh docs. If yes, can you put this on peer review (if it hasn't already been peer reviewed)?

geekspertise commented:
It has already been peer reviewed.

On Mon, Feb 25, 2019, 5:18 PM Vikram Goyal <notifications@github.com> wrote:

> @geekspertise <https://github.com/geekspertise> - I can't recall if we
> are going through the peer review process for ServiceMesh docs. If yes, can
> you put this on peer review (if it hasn't already been peer reviewed)?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift/openshift-docs/pull/13668#issuecomment-467206846>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/Anhd287UoDlz5aNlmquTpbaNVliDkMF1ks5vRGEpgaJpZM4bDqQ2>
> .
>

geekspertise commented:
@vikram-redhat When is this PR going to be merged? It was peer reviewed before I requested the merge.

kalexand-rh commented:
/cherrypick enterprise-3.11
